### PR TITLE
Insert blocks into remixed playlab levels to handle different level settings

### PR DESCRIPTION
**Pull request by balderdash**
Differing map layouts lead to a bunch of generated blocks, e.g.

Before Remix:
image

After Remix:
image

There would be even more if the original level also specified custom sprites/emotions/sizes/speeds.

### CODE REVIEW

File: apps/src/code-studio/initApp/project.js [hunk 1]
```diff
@@ -559,7 +559,15 @@ var projects = module.exports = {
         return null;
     }
   },
-  /**
+
+  canServerSideRemix() {
+    // The excluded app types need to make modifications to the project that
+    // apply to the remixed project, but should not be saved on the original
+    // project. See (Turtle|Studio).prepareForRemix().
+    return !['turtle', 'studio'].includes(appOptions.app);
```
Comment (joshlory) [inline apps/src/code-studio/initApp/project.js]: What happens if I try to remix one of these apps while logged out?

File: apps/src/studio/Item.js [hunk 2]
```diff
@@ -2,6 +2,7 @@ import Collidable from './collidable';
 import * as constants from './constants';
 import _ from 'lodash';
 import { ShakeActor } from './spriteActions';
+import Studio from './studio';
```
Comment (joshlory) [inline apps/src/studio/Item.js:5]: Is `Studio` already global?

Comment (balderdash) [reply apps/src/studio/Item.js:5]: Yes, but it seems to get set somewhere that doesn't run in these unit tests. This file is completely untested :crying_cat_face:

File: apps/src/code-studio/initApp/project.js (hunk 1 above)
Comment (balderdash) [reply apps/src/code-studio/initApp/project.js]: Ah, yeah it used to force you to log in and I broke that. Fixed by just disabling the last save.

**Pull request merged**